spark-publish/test/test.js

```diff
@@ -0,0 +1,124 @@
+import { publish } from '../index.js'
+import assert from 'node:assert'
+import { CID } from 'multiformats/cid'
+import pg from 'pg'
+
+const { DATABASE_URL } = process.env
+
+describe('unit', () => {
+  it('publishes', async () => {
+    const cid = 'bafybeicmyzlxgqeg5lgjgnzducj37s7bxhxk6vywqtuym2vhqzxjtymqvm'
+
+    const client = {
+      async query (_, params) {
+        assert.strictEqual(params[0], 1)
+
+        client.query = (_, params) => {
+          assert.strictEqual(params[0], cid)
+          assert.deepStrictEqual(params[1], [])
+        }
```

Comment:
I think we discussed this in the past. I have a bad experience with assertions inside mocked/stubbed functions. If the function is never called, then we never assert anything and the test passes, giving us a false sense of security.

A relatively easy solution:

```js
const dbQueryParams = []
client.query = (_, params) => { dbQueryParams.push(params) })
// ...

await publish({/*...*/}

assert.deepEquals(dbQueryParams, [
  [cid, []],
])
```

Reply:
good point, I was hoping to get your input on this